### DISCUSSION TITLE
C# Timer Error

### DISCUSSION THREAD

**Discussion by ghost** [community/community#65944] [Programming Help]
### Body

When I updated the code it stopped working 
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices.ComTypes;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Windows.Forms.VisualStyles;

namespace Sleeping_Dragons
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        public bool gameOver = false;
        public int lives = 10;
        public int score = 0;
        public int playerVelocityX = 5;
        public int playerVelocityY = 0;
        public bool goLeft;
        public void reloadGame()
        {
            this.gameOver = false;
            this.dungeonImg.Show();
            this.gameOverLabel.Hide();
            this.restartButton.Hide();
            this.livesLabel.Text = "Lives: " + lives;
            this.livesLabel.Show();
            this.scoreResult.Hide();
            this.score = 0;
            this.lives = 10;
            this.scoreLabel.Text = "Score: " + score;
            this.scoreLabel.Show();
            this.gameOverLabel.Hide();
            this.playerImg.Show();
        }
        private void Form1_Load(object sender, EventArgs e)
        {
            this.fireballImg.Hide();
            if (gameOver)
            {
                this.dungeonImg.Hide();
                this.gameOverLabel.Show();
                this.restartButton.Show();
                this.livesLabel.Hide();
                this.scoreLabel.Hide();
                this.scoreResult.Show();
                this.scoreResult.Text = "You scored: " + this.score;
                this.gameOverLabel.Show();
                this.playerImg.Hide();
            }
            else
            {
                this.dungeonImg.Hide();
                this.gameOverLabel.Hide();
                this.restartButton.Hide();
                this.livesLabel.Show();
                this.scoreLabel.Show();
                this.livesLabel.Text = "Lives: " + lives;
                this.scoreLabel.Text = "Score: " + score;
                this.gameOverLabel.Hide();
                this.scoreResult.Hide();
                this.playerImg.Show();
            }
            
        }

        private void dungeonImg_Click(object sender, EventArgs e)
        {

        }

        private void restartButton_Click(object sender, EventArgs e)
        {
            this.reloadGame();
        }

        private void Form1_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Left)
            {
                this.playerImg.Left = this.playerImg.Left - this.playerVelocityY;
            }
        }
        public void Timer1_Tick(object sender,
                                EventArgs e)
        {
            if (this.dragonImg.Visible)
            {
                this.fireballImg.Show();
            }
            else
            {
                this.fireballImg.Hide();
            }
            this.timer1.Interval = 1;
            this.timer1.Tick -= Timer1_Tick;
        }
    }
}

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by seyLu** [comment]
#### Hi @animeReview,

#### To format code in Github (for the sanity of other devs), wrap it in a code block, something like: 

<pre>
```cs
// triple backticks + language extension (cs -- in the case of c#)
// insert your code here
// close with triple backticks
```
</pre> 

#### Which would then render as:

```cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices.ComTypes;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Windows.Forms.VisualStyles;

namespace Sleeping_Dragons
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        public bool gameOver = false;
        public int lives = 10;
        public int score = 0;
        public int playerVelocityX = 5;
        public int playerVelocityY = 0;
        public bool goLeft;

        public void reloadGame()
        {
            this.gameOver = false;
            this.dungeonImg.Show();
            this.gameOverLabel.Hide();
            this.restartButton.Hide();
            this.livesLabel.Text = "Lives: " + lives;
            this.livesLabel.Show();
            this.scoreResult.Hide();
            this.score = 0;
            this.lives = 10;
            this.scoreLabel.Text = "Score: " + score;
            this.scoreLabel.Show();
            this.gameOverLabel.Hide();
            this.playerImg.Show();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            this.fireballImg.Hide();
            if (gameOver)
            {
                this.dungeonImg.Hide();
                this.gameOverLabel.Show();
                this.restartButton.Show();
                this.livesLabel.Hide();
                this.scoreLabel.Hide();
                this.scoreResult.Show();
                this.scoreResult.Text = "You scored: " + this.score;
                this.gameOverLabel.Show();
                this.playerImg.Hide();
            }
            else
            {
                this.dungeonImg.Hide();
                this.gameOverLabel.Hide();
                this.restartButton.Hide();
                this.livesLabel.Show();
                this.scoreLabel.Show();
                this.livesLabel.Text = "Lives: " + lives;
                this.scoreLabel.Text = "Score: " + score;
                this.gameOverLabel.Hide();
                this.scoreResult.Hide();
                this.playerImg.Show();
            }
        }

        private void dungeonImg_Click(object sender, EventArgs e) { }

        private void restartButton_Click(object sender, EventArgs e)
        {
            this.reloadGame();
        }

        private void Form1_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Left)
            {
                this.playerImg.Left = this.playerImg.Left - this.playerVelocityY;
            }
        }

        public void Timer1_Tick(object sender, EventArgs e)
        {
            if (this.dragonImg.Visible)
            {
                this.fireballImg.Show();
            }
            else
            {
                this.fireballImg.Hide();
            }
            this.timer1.Interval = 1;
            this.timer1.Tick -= Timer1_Tick;
        }
    }
}

```

#### And please also include the error message.

**Reply by seyLu** [reply]
#### When player presses left, it updates based on Y-velocity, is this the intended behaviour?

```cs
private void Form1_KeyDown(object sender, KeyEventArgs e)
{
    if (e.KeyCode == Keys.Left)
    {
        this.playerImg.Left = this.playerImg.Left - this.playerVelocityY;
    }
}
```

**Reply by seyLu** [reply]
#### When dealing with booleans, it is a better practice to prepend with `is`, like instead of `dragonImg.Visible`, change this to `dragonImg.IsVisible`:

```cs
public void Timer1_Tick(object sender, EventArgs e)
{
    if (this.dragonImg.Visible)
    {
        this.fireballImg.Show();
    }
    else
    {
        this.fireballImg.Hide();
    }
    this.timer1.Interval = 1;
    this.timer1.Tick -= Timer1_Tick;
}
```

#### Also the timer is ticking at 1 millisecond, which is very very frequent, is this the intended behaviour?

**Reply by seyLu** [reply]
#### Make sure everything is properly initialized, and ensure game state is managed correctly. If using VSCode, you can debug with breakpoints. Even a simple, `Console.WriteLine` goes a long way.

#### Please do provide the error message next time, so other devs would have more context and could better assist.

**Comment by ghost** [comment]
> When dealing with booleans, it is a bette yes it is. Sorry, Github didn't email me about any notifications for some reason

**Reply by ghost** [reply]
Windows Mail didn't send me a notification
